File: CONCEPTS/subsetting.py
import pandas as pd

# Create the DataFrame
data = {
    'GAME_ID': [1000, 1000, 1000, 1000, 2000, 2000, 2000, 2000],
    'TEAM_ID': [100, 100, 200, 200, 100, 100, 300, 300],
    'PLAYER_ID': [1, 2, 3, 4, 1, 2, 5, 6],
    'FGM': [10, 4, 2, 8, 10, 11, 8, 7],
    'FG3M': [12, 4, 6, 2, 4, 5, 10, 6],
    'FTM': [12, 7, 5, 7, 10, 4, 9, 3]
}
df = pd.DataFrame(data)

print('--- 1 --- # Selecting a single column as a DataFrame')
a = df[['GAME_ID']] 
print(type(a))
print(a)

print('\n--- 2 --- # Selecting a specific scalar value')
b = df.iloc[0, 0]
print(type(b))
print(b)

print('\n--- 3 --- # Selecting a single row')
c = df.iloc[[0]]
print(type(c))
print(c)

print('\n--- 4 --- # Selecting a single column as a Series')
d = df.iloc[:, 0]
print(type(d))
print(d)

print('\n--- 5 --- # Extracting column values (equivalent to [[1]])')
x = df['GAME_ID']
print(type(x))
print(x)

File: CONCEPTS/subsetting.R
# df <- tibble::tribble(
#   ~GAME_ID, ~TEAM_ID, ~PLAYER_ID, ~FGM, ~FG3M, ~FTM,
#       1000,      100,          1,   10,    12,   12,
#       1000,      100,          2,    4,     4,    7,
#       1000,      200,          3,    2,     6,    5,
#       1000,      200,          4,    8,     2,    7,
#       2000,      100,          1,   10,     4,   10,
#       2000,      100,          2,   11,     5,    4,
#       2000,      300,          5,    8,    10,    9,
#       2000,      300,          6,    7,     6,    3
# )

df <- data.frame(
  GAME_ID = c(1000, 1000, 1000, 1000, 2000, 2000, 2000, 2000),
  TEAM_ID = c(100, 100, 200, 200, 100, 100, 300, 300),
  PLAYER_ID = c(1, 2, 3, 4, 1, 2, 5, 6),
  FGM = c(10, 4, 2, 8, 10, 11, 8, 7),
  FG3M = c(12, 4, 6, 2, 4, 5, 10, 6),
  FTM = c(12, 7, 5, 7, 10, 4, 9, 3)
)

print('--- 1 ---')
a <- df[1]
print(class(a))
print(a)

print('--- 2 ---')
b <- df[1,1]
print(class(b))
print(b)

print('--- 3 ---')
c <- df[1,]
print(class(c))
print(c)

print ('--- 4 ---')
d <- df[,1]
print(class(d))
print(d)

print('--- 5 ---')
x <- df[[1]]
print(class(x))
print(x)

